sdk/src/sysvars/rent.rs

```diff
@@ -174,27 +177,88 @@ impl Rent {
         let bytes = data_len as u64;
-        let exemption_threshold = u64::from_le_bytes(self.exemption_threshold);
 
-        match exemption_threshold {
-            SIMD0194_EXEMPTION_THRESHOLD => {
-                (ACCOUNT_STORAGE_OVERHEAD + bytes) * self.lamports_per_byte
-            }
-            CURRENT_EXEMPTION_THRESHOLD => {
-                2 * (ACCOUNT_STORAGE_OVERHEAD + bytes) * self.lamports_per_byte
-            }
-            _ => {
-                #[cfg(not(target_arch = "bpf"))]
-                {
-                    (((ACCOUNT_STORAGE_OVERHEAD + bytes) * self.lamports_per_byte) as f64
-                        * f64::from_le_bytes(self.exemption_threshold)) as u64
-                }
-                #[cfg(target_arch = "bpf")]
-                panic!("Floating-point operations are not supported on BPF targets");
+        // There are two cases where it is possible to avoid floating-point
+        // operations:
+        //
+        //   1)  exemption threshold is `1.0` (the SIMD-0194 default)
+        //   2)  exemption threshold is `2.0` (the current default)
+        //
+        // In all other cases, perform the full calculation using floating-point
+        // operations. Note that on BPF targets, floating-point operations are
+        // not supported, so panic in that case.
+        if self.exemption_threshold == SIMD0194_EXEMPTION_THRESHOLD {
+            (ACCOUNT_STORAGE_OVERHEAD + bytes) * self.lamports_per_byte
+        } else if self.exemption_threshold == CURRENT_EXEMPTION_THRESHOLD {
+            2 * (ACCOUNT_STORAGE_OVERHEAD + bytes) * self.lamports_per_byte
+        } else {
+            #[cfg(not(target_arch = "bpf"))]
+            {
+                (((ACCOUNT_STORAGE_OVERHEAD + bytes) * self.lamports_per_byte) as f64
+                    * f64::from_le_bytes(self.exemption_threshold)) as u64
             }
+            #[cfg(target_arch = "bpf")]
+            panic!("Floating-point operations are not supported on BPF targets");
+        }
+    }
+
+    /// Calculates the minimum balance for rent exemption.
+    ///
+    /// This method avoids floating-point operations when the `exemption_threshold`
+    /// is the default value.
+    ///
+    /// # Arguments
+    ///
+    /// * `data_len` - The number of bytes in the account
+    ///
+    /// # Returns
+    ///
+    /// The minimum balance in lamports for rent exemption.
+    ///
+    /// # Errors
+    ///
+    /// Returns `ProgramError::InvalidArgument` if `data_len` exceeds the maximum permitted
+    /// data length.
+    #[inline(always)]
+    pub fn try_minimum_balance(&self, data_len: usize) -> Result<u64, ProgramError> {
+        if data_len as u64 > MAX_PERMITTED_DATA_LENGTH {
+            Err(ProgramError::InvalidArgument)
+        } else {
+            // SAFETY: The `data_len` is validated to be lower than the maximum permitted
+            // data length.
+            Ok(unsafe { self.minimum_balance_unchecked(data_len) })
```

Comment:
Do we want to worry about the case where `lamports_per_byte` is huge? Because technically this can still overflow

Reply:
We can add the `if` statements for each "known" value of the exempt threshold. For the float fallback there is no way to do the checked math, so we can leave as it is. The `if` statements will probably be cheaper than the checked math.